### ISSUE TITLE
Consider expanding the HttpClient to support a query parameter and fragment builders.

### ISSUE THREAD

**Issue by flyinprogrammer** [reactor/reactor-netty#753]
Today we make it super easy to create an `HttpClient` with headers that are set, or even conditionally set, for the lifetime of the `HttpClient` - which is super cool!

https://github.com/reactor/reactor-netty/blob/master/src/main/java/reactor/netty/http/client/HttpClient.java#L611-L624

However, there are cases where APIs use query parameters to achieve clever things. For example, (and I acknowledge this is weak) the Consul API allows (and now discourages) the use of an [API security token to be passed to Consul via URI Query Parameter](https://www.consul.io/api/index.html#authentication). Today, in order to support this we must use a delegate pattern and override the `uri` methods of an `HttpClient` class to inject the token.

If we had helper builder functions on the client, similar to the ones we have for headers, the delegate would be unnecessary.

Similarly a URI Fragment builder could also be added.

**Comment by odravison** [comment]
@violetagg , That is what I need. Do you know if still need help with this?

**Comment by violetagg** [comment]
@odravison we have `uri(URI)` isn't this helpful?

**Comment by larry-cable** [comment]
I agree with @flyinprogrammer it would be convenient if the HttpClient had a similar "builder" method for Headers or perhaps an overloaded uri() with a Map<String, String> of Query parameters that would be properly encoded and appended to the uri?

**Comment by LivingLikeKrillin** [comment]
Hi @violetagg,

I’m interested in contributing to this feature (#753) by adding a query parameter and fragment builder to `HttpClient`, similar to the existing builder methods for headers.

I believe this will make it much easier to build dynamic URIs and reduce the need for boilerplate or custom delegation patterns.

If this is something you'd still like help with, please let me know any preferences or design directions you have in mind.

Happy to help!

**Comment by violetagg** [comment]
@LivingLikeKrillin We have this PR https://github.com/reactor/reactor-netty/pull/2696 which we would like to include in 1.3 version
